redis-snapshot-replicator/functions/shipper.py

```diff
@@ -20,9 +20,13 @@ def lambda_handler(event, context):
 
     source = boto3.client('elasticache')
 
-    for instance in instances.split(','):
-        source_snaps = source.describe_snapshots(CacheClusterId=instance)['Snapshots']
-        source_snap = sorted(source_snaps, key=byTimestamp, reverse=True)[0]['SnapshotName']
+    message = event['Records'][0]['Sns']['Message']
+
+    if "Snapshot succeeded" in message
+
+        for instance in instances.split(','):
```

Comment:
Can't you extract the snapshot id from the sns event? Getting the last snapshot instead of the one that actually triggered the event doesn't feel a very predictable system IMO

Reply:
makes sense although it might be tricky, let me try

Reply:
and it worked! thanks